File: backend/Dtos/GetDtos/FilteringRequest.cs
namespace backend.Dtos.GetDtos
{
    public class FilteringRequest
    {
        public string? Name {set; get;}
        public string? Author {set; get;}
        public string? Category {set; get;}
        public DateTime? FromDate {set; get;}
        public DateTime? ToDate {set; get;}
        public string? SortField {set; get;}
        public string? SortDirection {set; get;}
        public int PageNumber {set; get;}
        public int PageSize {set; get;}
    }
}

File: frontend/src/app/interfaces/common/FilteringRequest.ts
export interface FilteringRequest
{
  name?: string;
  author?: string;
  category?: string;
  fromDate?: Date;
  toDate?: Date;
  sortField?: string;
  sortDirection?: string;
  IsDeleted? :boolean;
  pageNumber: number;
  pageSize: number;
}
